src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconService.kt

```diff
@@ -115,47 +115,57 @@ class ActivitiesReconService(
   private suspend fun publishTelemetry(
     type: String,
     prisonId: String,
-    compareResults: CompareBookingsResult,
+    differences: CompareBookingsDifferences,
     date: LocalDate? = null,
   ) {
-    if (compareResults.noDifferences()) {
+    val differencesWithDetails = with(differences.all()) {
+      if (isNotEmpty()) nomisApiService.getPrisonerDetails(this) else listOf()
+    }
+
+    val ignoredBookings = differencesWithDetails
+      .filter { it.location != prisonId }
+      .onEach {
+        log.info("Ignoring failed $type reconciliation for booking ${it.bookingId} at prison $prisonId as the prisoner is now at location ${it.location}")
+      }
+      .map { it.bookingId }
+
+    // Publish telemetry for failures we're not ignoring
+    differences.all()
+      .filterNot { it in ignoredBookings }
+      .sorted()
+      .map { differencesWithDetails.first { details -> details.bookingId == it } }
+      .forEach {
+        telemetryClient.trackEvent(
+          "activity-$type-reconciliation-report-failed",
+          mutableMapOf(
+            "prison" to prisonId,
+            "type" to differences.differenceType(it.bookingId),
+            "bookingId" to it.bookingId.toString(),
+            "offenderNo" to it.offenderNo,
+            "location" to it.location,
+          ).apply {
+            date?.also { this["date"] = "$it" }
+          },
+        )
+      }
+
+    // Publish success telemetry if all failures were ignored or there were no failures
+    if (differences.all().size - ignoredBookings.size == 0) {
       telemetryClient.trackEvent(
         "activity-$type-reconciliation-report-success",
         mutableMapOf("prison" to prisonId).apply {
           date?.also { this["date"] = "$it" }
         },
       )
```

Comment:
This is the fix - previously if all failures were ignored then we didn't publiksh the success telemetry, but we should.